### PR TITLE
quick_sort.cpp: Add QuickSort C++

### LINKED ISSUE #250: [Algo] Quick Sort [C++]
Hi!
Can you assign this to me, please?

### PR DESCRIPTION
**Pull request by joao-p-marques**
Closes https://github.com/NITSkmOS/Algorithms/issues/250

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

Fixes #250 

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [X] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [X] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [X] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [X] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [X] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [X] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [X] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

### CODE REVIEW

File: quicksort/Cpp/quick_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
+#include <iostream>
+
+using std::cout;
+
+void quick_sort(int a[], int left, int right) {
+
```
Comment (sangamcse) [inline quicksort/Cpp/quick_sort.cpp]: Redundant blank line at the start of a code block should be deleted.  [whitespace/blank_line] [2]

*Origin: CPPLintBear, Section: `all.cpplint`.*

File: quicksort/Cpp/quick_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            j--; //carry on, this part already sorted
+        if (i <= j) { // a[i]>=pivot and a[j]<=pivot
+            //swap a[i] with a[j]
+            int tmp = a[i];
+            a[i] = a[j];
+            a[j] = tmp;
+            i++;
+            j--;
+        }
+      };
```
Comment (sangamcse) [inline quicksort/Cpp/quick_sort.cpp]: You don't need a ; after a }  [readability/braces] [4]

*Origin: CPPLintBear, Section: `all.cpplint`.*

File: quicksort/Cpp/quick_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+      if (left < j)
+        quick_sort(a, left, j);
+      if (i < right)
+        quick_sort(a, i, right);
+
```
Comment (sangamcse) [inline quicksort/Cpp/quick_sort.cpp]: Redundant blank line at the end of a code block should be deleted.  [whitespace/blank_line] [3]

*Origin: CPPLintBear, Section: `all.cpplint`.*

File: quicksort/Cpp/quick_sort.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+int main(void){
```
Comment (sangamcse) [inline quicksort/Cpp/quick_sort.cpp]: Missing space before {  [whitespace/braces] [5]

*Origin: CPPLintBear, Section: `all.cpplint`.*

File: quicksort/Cpp/quick_sort.cpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    int a[10] = { 10, 7, 9, 4, 6, 2, 5, 6, 84, 12};
+    int a_size = sizeof(a)/sizeof(a[0]);
+
+    quick_sort(a, 0, a_size-1);
+
+    for(int i=0; i<a_size; i++){
```
Comment (sangamcse) [inline quicksort/Cpp/quick_sort.cpp]: Missing spaces around =  [whitespace/operators] [4]

*Origin: CPPLintBear, Section: `all.cpplint`.*

Comment (sangamcse) [inline quicksort/Cpp/quick_sort.cpp]: Missing spaces around <  [whitespace/operators] [3]

*Origin: CPPLintBear, Section: `all.cpplint`.*

Comment (sangamcse) [inline quicksort/Cpp/quick_sort.cpp]: Missing space before ( in for(  [whitespace/parens] [5]

*Origin: CPPLintBear, Section: `all.cpplint`.*

Comment (sangamcse) [inline quicksort/Cpp/quick_sort.cpp]: Missing space before {  [whitespace/braces] [5]

*Origin: CPPLintBear, Section: `all.cpplint`.*

File: quicksort/Cpp/quick_sort.cpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        cout << a[i] << " ";
+    }
+}
```
Comment (sangamcse) [inline quicksort/Cpp/quick_sort.cpp]: Could not find a newline character at the end of the file.  [whitespace/ending_newline] [5]

*Origin: CPPLintBear, Section: `all.cpplint`.*

File: quicksort/Cpp/quick_sort.cpp [hunk 7]
```diff
@@ -0,0 +1,42 @@
+
```
Comment (sangamcse) [inline quicksort/Cpp/quick_sort.cpp]: Redundant line

**Comment by joao-p-marques** [comment]
Changes made.

**Comment by sangamcse** [comment]
@joao-p-marques checks are still failing

**Pull request closed without merging**